test/unit/role_test.rb

```diff
@@ -63,6 +63,16 @@ class RoleTest < ActiveSupport::TestCase
         assert_equal Role::BUILTIN_DEFAULT_ROLE, role.builtin
       end
     end
+
+    should "have zero or one builtin" do
+      role = Role.new(:name => 'role name')
+      [0, 1, 2].each do |i|
```

Comment:
I'd recommend to split these to two different should block - two different test cases, for 0 and 1 you'd verify that `role.must_be :valid?`, in second test for 2 you'd verify that role is invalid, ideally by `role.wont_be :valid?`
also I don't think role.save is necessary, validations should fail on calling `valid?` when it's being tested